P0009/P0009.bs

```diff
@@ -897,60 +903,46 @@ template<ptrdiff_t... OtherExtents>
+	     for each `r` in range `[0,rank())`.
+* *Effects:* For each `r` in range `[0,rank())` where `static_extent(r)==std::dynamic_extent`, initializes `dynamic_extent[dynamic_rank(r)]` with `other.extent(r)`.
 * *Throws:* Nothing.
 
 <br/>
 
 ```c++
 template<class... IndexType>
-constexpr extents(IndexType... dynamic_extents) noexcept;
+constexpr extents(IndexType... dynamic) noexcept;
 ```
 
-* *Requires:* `(dynamic_extents>=0) && ...` is `true`.
 * *Constraints:* 
-    * `is_convertible_v<IndexType, index_type> && ...` is `true`, and
-    * `sizeof...(dynamic_extents)==rank_dynamic()` is `true`.
-* *Effects:* Aggregate-initializes `dynamic_extents_` to `{dynamic_extents...}`.
-* *Ensures:* `extent(`*DynamicRank[i]*`)` is equal to the *i*th entry in the parameter pack `dynamic_extents`.
+    * `(is_convertible_v<IndexType, index_type> && ...)` is `true` and
+    * `sizeof...(dynamic)==rank_dynamic()` is `true`.
+* *Expects:* `((dynamic_extents>=0) && ...)` is `true`.
+* *Effects:* Initializes `dynamic_extents_` with `dynamic...`.
 
 ```c++
-template<class IndexType, size_t rank_dynamic>
-constexpr extents(const array<IndexType, rank_dynamic> & dynamic_extents) noexcept;
+template<class IndexType>
+constexpr extents(const array<IndexType, rank_dynamic()> & dynamic) noexcept;
 ```
 
-* *Requires:* Let <math>D</math> equal `rank_dynamic()`.
-             <math>&forall; `i` &isin; [0, D)</math>, 
-             `dynamic_extents[i] >= 0` is `true`.
 * *Constraints:* `is_convertible_v<IndexType, index_type>` is `true`.
-* *Effects:* Initializes `dynamic_extents_` with `dynamic_extents`.
-* *Ensures:* `extent(`*DynamicRank[i]*`)` equals `dynamic_extents[i]`.
+* *Expects:* `dynamic[i]>=0` for all `i` in range `[0,rank_dynamic())`.
+* *Effects:* Initializes `dynamic_extents_` with `dynamic`.
 
 ```c++
 template<ptrdiff_t... OtherExtents>
 constexpr extents& operator=(const extents<OtherExtents...>& other);
 ```
 
-* *Requires:* Let <math>R</math> equal `rank()`.
-              <math>&forall; `r` &isin; [0, R)</math>, 
-              `static_extent(r) != dynamic_extent || static_extent(r) == other.extent(r)` is `true`.
-* *Constraints:* `sizeof...(Extents)==sizeof...(OtherExtents)` is `true`.
-* *Effects:* Let <math>R</math> equal `rank()`.
-              <math>&forall; `r` &isin; [0, R)</math>,
-             if `static_extent(r)` equals `dynamic_extent`, 
-             assigns `dynamic_extents_[`*DynamicRank[*`r`*]*`]` to `other.extent(r)`.
-* *Ensures:* `*this==other` is `true`.
-* *Returns:* `*this`.
-* *Throws:* Nothing.
+* *Constraints:* `sizeof...(StaticExtents)` equals `sizeof...(OtherExtents)`.
+* *Expects:* `static_extent(r)==std::dynamic_extent` or
+             `static_extent(r)==other.extent(r)`
+	     for all `r` in range `[0,rank())`.
+* *Effects* For each `r` in the range `[0,rank())` where `static_extent(r)==std::dynamic_extent`, assigns `dynamic_extent[dynamic_rank(r)]` to `other.extent(r)`.
```

Comment:
Need `dynamic_extents_` not `dynamic_extent` here.  Also, should assign `other.extent(r)` to `dynamic_extents_[dynamic_rank(r)]`, not the other way.

Reply:
Need a colon after "Effects".